File: backend/src/main/java/com/senior/project/backend/domain/Club.java
package com.senior.project.backend.domain;

import lombok.*;

import java.util.Date;
import java.util.UUID;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Temporal;
import jakarta.persistence.TemporalType;

@Entity
@Data
@ToString
@NoArgsConstructor
@AllArgsConstructor
@Generated
public class Club {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID clubId;

    private UUID clubName;

    @Temporal(value = TemporalType.TIMESTAMP)
    private Date startDate;

    @Temporal(value = TemporalType.TIMESTAMP)
    private Date endDate;
}

File: frontend/crd/src/domain/Club.ts
export interface ClubJSON{
    id: string;
    clubName: string;
    startDate: Date;
    endDate: Date;
}

export class Club{
    constructor(json: ClubJSON){
        this.id = json.id;
        this.clubName = json.clubName;
        this.startDate = new Date(json.startDate);
        this.endDate = new Date(json.endDate);
    }

    id: string;
    clubName: string;
    startDate: Date;
    endDate: Date;
}